src/plots/PlotlyPlot.tsx

```diff
@@ -1,76 +1,140 @@
-import React, { lazy, Suspense, useMemo } from 'react';
+import React, { lazy, Suspense, useMemo, CSSProperties } from 'react';
 import { PlotParams } from 'react-plotly.js';
+import { legendSpecification } from '../utils/plotly';
+import Spinner from '../components/Spinner';
+import { PlotLegendAddon, PlotSpacingAddon } from '../types/plots/addOns';
+import { UnionOfPlotDataTypes } from '../types/plots';
+import { LayoutLegendTitle } from '../types/plotly-omissions';
 
-type Margin = {
-  l: number;
-  r: number;
-  t: number;
-  b: number;
-};
-
-// set props for legend position
-type legendProp = {
-  x?: number;
-  y?: number;
-  // xanchor is for positioning legend inside plot
-  xanchor?: 'auto' | 'center' | 'left' | 'right';
-  orientation?: 'h' | 'v' | undefined;
-};
-
-export interface PlotProps {
-  width?: number;
-  height?: number;
-  margin?: Partial<Margin>;
-  staticPlot?: boolean;
-  showModebar?: boolean | 'hover';
-  // add legend prop for positioning
-  legend?: legendProp;
-  // show a loading spinner on top of the plot
+export interface PlotProps<T extends UnionOfPlotDataTypes> {
```

Comment:
This part gives me pause. I don't think it is necessary to specify `T extends UnionOfPlotDataTypes`. I think `T` can simply be generic.

```suggestion
export interface PlotProps<T> {
```

Reply:
Thanks - I thought it would be helpful to constrain the XxxData types to ones we knew about, but I suppose it's a pain to maintain UnionOfPlotDataTypes.  The latter could probably be retired at some point (it is mainly a usePlotControls thing, I think) but not now.